File: frontend/src/types/comment.ts
export interface Comment {
  id: number
  comment_id: string
  parent_id: string | null
  author_name: string
  author_channel_id: string | null
  text: string
  like_count: number
  published_at: string | null
  reply_count: number
}

export interface CommentListResponse {
  comments: Comment[]
  total: number
  page: number
  limit: number
  has_more: boolean
}

File: backend/app/schemas/comment.py
from uuid import UUID
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class CommentResponse(BaseModel):
    """Single comment for display."""
    id: int
    comment_id: str
    parent_id: Optional[str] = None
    author_name: str
    author_channel_id: Optional[str] = None
    text: str
    like_count: int = 0
    published_at: Optional[datetime] = None
    reply_count: int = 0

    class Config:
        from_attributes = True


class CommentListResponse(BaseModel):
    """Paginated comment list."""
    comments: List[CommentResponse]
    total: int
    page: int
    limit: int
    has_more: bool
